From ac59e93d15abab3396f81be83285a0726f6a1df5 Mon Sep 17 00:00:00 2001
From: George Kiagiadakis <george.kiagiadakis@collabora.com>
Date: Fri, 20 Apr 2012 17:13:52 +0300
Subject: [PATCH 2/4] Initialize structures to version 1.1.2

---
 omx/gstomx.h |    1 +
 1 files changed, 1 insertions(+), 0 deletions(-)

diff --git a/omx/gstomx.h b/omx/gstomx.h
index 5b222e2..189354b 100644
--- a/omx/gstomx.h
+++ b/omx/gstomx.h
@@ -35,6 +35,7 @@ G_BEGIN_DECLS
   (st)->nSize = sizeof (*(st)); \
   (st)->nVersion.s.nVersionMajor = 1; \
   (st)->nVersion.s.nVersionMinor = 1; \
+  (st)->nVersion.s.nRevision = 2; \
 } G_STMT_END
 
 /* Different hacks that are required to work around
-- 
1.7.8.5

