From ac5f1b7ff5adef158a9c2976f7ce8cc7f9eb381e Mon Sep 17 00:00:00 2001
From: George Kiagiadakis <george.kiagiadakis@collabora.com>
Date: Sat, 21 Apr 2012 13:58:51 +0300
Subject: [PATCH 3/4] Use OMX_COLOR_FormatYUV420PackedPlanar for I420

---
 omx/gstomxvideodec.c |    4 +++-
 1 files changed, 3 insertions(+), 1 deletions(-)

diff --git a/omx/gstomxvideodec.c b/omx/gstomxvideodec.c
index ae102f6..5f9d20e 100644
--- a/omx/gstomxvideodec.c
+++ b/omx/gstomxvideodec.c
@@ -670,6 +670,7 @@ gst_omx_video_dec_loop (GstOMXVideoDec * self)
 
     switch (port_def.format.video.eColorFormat) {
       case OMX_COLOR_FormatYUV420Planar:
+      case OMX_COLOR_FormatYUV420PackedPlanar:
         state->format = GST_VIDEO_FORMAT_I420;
         break;
       case OMX_COLOR_FormatYUV420SemiPlanar:
@@ -1005,6 +1006,7 @@ gst_omx_video_dec_negotiate (GstOMXVideoDec * self)
     if (err == OMX_ErrorNone) {
       switch (param.eColorFormat) {
         case OMX_COLOR_FormatYUV420Planar:
+        case OMX_COLOR_FormatYUV420PackedPlanar:
           gst_caps_append_structure (comp_supported_caps,
               gst_structure_new ("video/x-raw-yuv",
                   "format", GST_TYPE_FOURCC, GST_MAKE_FOURCC ('I', '4', '2',
@@ -1048,7 +1050,7 @@ gst_omx_video_dec_negotiate (GstOMXVideoDec * self)
 
   switch (format) {
     case GST_VIDEO_FORMAT_I420:
-      param.eColorFormat = OMX_COLOR_FormatYUV420Planar;
+      param.eColorFormat = OMX_COLOR_FormatYUV420PackedPlanar;
       break;
     case GST_VIDEO_FORMAT_NV12:
       param.eColorFormat = OMX_COLOR_FormatYUV420SemiPlanar;
-- 
1.7.8.5

